apps/src/codebridge/Codebridge.tsx

```diff
@@ -31,6 +32,7 @@ type CodebridgeProps = {
   startSource: ProjectSources;
   onRun?: OnRunFunction;
   onStop?: () => void;
+  projectVersion: number;
```

Comment:
The old syncing issues were due to trying to keep track of whether the project had changed _internally_ inside the Codebridge component. Instead, we defer it to externally - in the last rev, the `useSource` hook got a `codeBridgeKey` which was used to re-key the component.

Now, that same incrementing value is used as a `projectVersion`, and `<CodeBridge/>` looks to see if it has changed. If so, it'll replace the internal project but won't re-key anything, so it'll trigger fewer re-renders.

There may still be a way to manage this all internally via a wrapper or something, but we can circle back to it.